tests/test-class-amp-options-manager.php

```diff
@@ -0,0 +1,151 @@
+		);
+
+		// Test analytics validation with missing fields.
+		AMP_Options_Manager::update_option( 'analytics', array(
+			'bad' => array(),
+		) );
+		$errors = get_settings_errors( AMP_Options_Manager::OPTION_NAME );
+		$this->assertEquals( 'missing_analytics_vendor_or_config', $errors[0]['code'] );
+		$wp_settings_errors = array();
+
+		// Test analytics validation with bad JSON.
+		AMP_Options_Manager::update_option( 'analytics', array(
+			'__new__' => array(
+				'type'   => 'foo',
+				'config' => 'BAD',
+			),
+		) );
+		$errors = get_settings_errors( AMP_Options_Manager::OPTION_NAME );
+		$this->assertEquals( 'invalid_analytics_config_json', $errors[0]['code'] );
+		$wp_settings_errors = array();
+
+		// Test analytics validation with good fields.
+		AMP_Options_Manager::update_option( 'analytics', array(
+			'__new__' => array(
+				'type'   => 'foo',
+				'config' => '{"good":true}',
+			),
+		) );
+		$this->assertEmpty( get_settings_errors( AMP_Options_Manager::OPTION_NAME ) );
+
+		// Test analytics validation with duplicate check.
+		AMP_Options_Manager::update_option( 'analytics', array(
+			'__new__' => array(
+				'type'   => 'foo',
+				'config' => '{"good":true}',
+			),
+		) );
+		$errors = get_settings_errors( AMP_Options_Manager::OPTION_NAME );
+		$this->assertEquals( 'duplicate_analytics_entry', $errors[0]['code'] );
+		$wp_settings_errors = array();
+
+		// Confirm format of entry ID.
+		$entries = AMP_Options_Manager::get_option( 'analytics' );
+		$entry   = current( $entries );
+		$id      = substr( md5( $entry['type'] . $entry['config'] ), 0, 12 );
+		$this->assertArrayHasKey( $id, $entries );
+		$this->assertEquals( 'foo', $entries[ $id ]['type'] );
+		$this->assertEquals( '{"good":true}', $entries[ $id ]['config'] );
+
+		// Confirm adding another entry works.
+		AMP_Options_Manager::update_option( 'analytics', array(
+			'__new__' => array(
+				'type'   => 'bar',
+				'config' => '{"good":true}',
+			),
+		) );
+		$entries = AMP_Options_Manager::get_option( 'analytics' );
+		$this->assertCount( 2, AMP_Options_Manager::get_option( 'analytics' ) );
+		$this->assertArrayHasKey( $id, $entries );
+
+		// Confirm updating an entry works.
+		AMP_Options_Manager::update_option( 'analytics', array(
+			$id => array(
+				'id'     => $id,
+				'type'   => 'foo',
+				'config' => '{"very_good":true}',
+			),
+		) );
+		$entries = AMP_Options_Manager::get_option( 'analytics' );
+		$this->assertEquals( 'foo', $entries[ $id ]['type'] );
+		$this->assertEquals( '{"very_good":true}', $entries[ $id ]['config'] );
+
+		// Confirm deleting an entry works.
+		AMP_Options_Manager::update_option( 'analytics', array(
+			$id => array(
+				'id'     => $id,
+				'type'   => 'foo',
+				'config' => '{"very_good":true}',
+				'delete' => true,
+			),
+		) );
+		$entries = AMP_Options_Manager::get_option( 'analytics' );
+		$this->assertCount( 1, $entries );
+		$this->assertArrayNotHasKey( $id, $entries );
```

Comment:
Would be good to cleanup the option here not to leave traces behind which might affect other tests.

Reply:
Unnecessary because tests in core automatically clean up after themselves: https://github.com/WordPress/wordpress-develop/blob/1c2b0762c7f1a91394c633ae812505919bd297ed/tests/phpunit/includes/testcase.php#L102-L182